### PR TITLE
Refactor upload object params

### PR DESCRIPTION
**Pull request by bamader**
This PR addresses the work scoped in https://app.clickup.com/t/3c06bt8. The function signature has been modified to put the data parameter (`message`) first, and message now accepts either a string or a dictionary to eliminate the need for maintaining separate upload functions.

### CODE REVIEW

**Review by spencer-skylight — CHANGES_REQUESTED** [review]
This looks great to me. 

Just one little suggestion... 
The `upload_object` docstring description in `BaseCloudContainerConnection` should be updated to match the one you changed in `AzureCloudContainerConnection`.

File: tests/cloud/test_cloud.py [hunk 1]
```diff
@@ -265,9 +265,9 @@ def test_upload_object(mock_get_client):
     )
 
     phdi_container_client.upload_object(
+        object_content,
```
Comment (spencer-skylight) [inline tests/cloud/test_cloud.py:268]: We should probably test both string and dict uploads.

Comment (bamader) [reply tests/cloud/test_cloud.py]: Oh yeah great call, totally meant to do that

**Comment by bamader** [comment]
Documentation updated and test added

**Review by spencer-skylight — APPROVED** [review]
LGTM!

**Pull request merged**